### PR TITLE
cleanup(bigtable): deprecate admin async functions

### LINKED ISSUE #2857: Use [[deprecated]] attributes.
These attributes are available since C++14.

We could add a macro or wait until dropping support for C++11.

### PR DESCRIPTION
**Pull request by coryan**
All the `Async*` functions in bigtable are marked as experimental. We do
not think the admin APIs need async versions, so it is time to mark them
as deprecated.

I wanted to use `ABSL_DEPRECATED()`, but that macro only works for
Clang, and we have users on GCC and MSVC to worry about.

Part of the work for #5923. Fixes #2857 

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5924)

### CODE REVIEW

**Review by devjgm — APPROVED** [review]

**Review by devjgm — APPROVED** [review]

**Comment by devbww** [comment]
> I wanted to use ABSL_DEPRECATED(), but that macro only works for
> Clang, and we have users on GCC and MSVC to worry about.

A rhetorical question, but doesn't Abseil have GCC and MSVC users to worry about too?

**Comment by coryan** [comment]
> > I wanted to use ABSL_DEPRECATED(), but that macro only works for
> > Clang, and we have users on GCC and MSVC to worry about.
> 
> A rhetorical question, but doesn't Abseil have GCC and MSVC users to worry about too?

Yes, and I think we should contribute the support for other compilers.  I did not want to wait for a full cycle of Abseil + gRPC releases before starting to use it.  We can replace the macro when they are ready.

/FYI @derekmauro

File: google/cloud/internal/attributes.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+#if defined(GOOGLE_CLOUD_CPP_DEPRECATED)
+#error "Applications should not define GOOGLE_CLOUD_CPP_DEPRECATED"
+#elif GOOGLE_CLOUD_CPP_DISABLE_DEPRECATION_WARNINGS
+// Do nothing, the default definition effectively
+#elif GOOGLE_CLOUD_CPP_CPP_VERSION >= 201402L
+#define GOOGLE_CLOUD_CPP_DEPRECATED(message) [[deprecated(message)]]
+#elif defined(_MSC_VER)
+#define GOOGLE_CLOUD_CPP_DEPRECATED(message) __declspec(deprecated(message))
+#elif defined(__clag__) || defined(__GNUC__)
```
Comment (derekmauro) [inline google/cloud/internal/attributes.h:43]: I think you mean `__clang__`.

**Review by derekmauro — COMMENTED** [review]
I didn't realize our implementation of ABSL_DEPRECATED was stale. Thanks for letting me know, I'll look into updating it.

File: google/cloud/internal/attributes.h (hunk 1 above)
Comment (coryan) [reply google/cloud/internal/attributes.h]: Fixed, thanks!

**Pull request merged**